openapi.yml

```diff
@@ -45,26 +65,44 @@ paths:
           in: query
           schema:
             $ref: "#/components/schemas/Address"
+        - name: sellAmount
+          in: query
+          schema:
+            $ref: "#/components/schemas/TokenAmount"
+        - name: buyAmount
+          in: query
+          schema:
+            $ref: "#/components/schemas/TokenAmount"
+        - name: orderType
+          in: query
+          schema:
+            $ref: "#/components/schemas/OrderType"
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken or buyToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
       type: integer
+    OrderType:
+      description: Is this a buy order or sell order?
+      type: string
+      enum: [buy, sell]
     OrderNew:
```

Comment:
Yeah, or another alternative would be `OrderCore` + `OrderEnrichement` = `Order`?

Reply:
There must be two types because only some information is needed when creating an order but more is returned when querying. I don't have a strong preference on what they are called.

Reply:
For me was a nit, so go with a name you feel comfortable. This ones won't affect the endpoint itself :) 
My reasoning is, when they are just parameters for creating an entity, I just use a convention of "Creation" or "CreationParams", because "New" means "recent" (it could even be mistaken for order modification). With creation or creationParams I would like to suggest is the data required to create an order.

That's my reasoning, but go with what you feel comfortable.

Reply:
I agree that this should be called `Order` to match the smart contract and spec. I think the order with additional signature/user data should be called `SignedOrder`.

Reply:
I'm happy anyways, the objects won't affect the endpoint signature. It sounded good to me what u first said of calling what the user `POST` a `SignedOrder`